### PR TITLE
Move release process to goreleaser

### PR DESCRIPTION
**Pull request by rohitagg2020**
Move release process to goreleaser.

Signed-off-by: rohitagg2020 <rohit.aggarwal2020@gmail.com>

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

##### Review Checklist:

- [ ] Follows the [developer guidelines](https://carvel.dev/shared/docs/latest/development_guidelines/)
- [ ] Relevant tests are added or updated
- [ ] Relevant docs in this repo added or updated
- [ ] Relevant carvel.dev docs added or updated in a separate PR and there's
  a link to that PR
- [ ] Code is at least as readable and maintainable as it was before this
  change

#### Additional documentation e.g., Proposal, usage docs, etc.:

```docs

```

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -20,14 +23,15 @@ jobs:
...
-      - name: Build kapp Script
-        run: |
-          ./hack/build-binaries.sh
-          shasum -a 256 ./kapp-* | tee checksums.txt
-          echo "# :open_file_folder: Files Checksum" | tee checksums-formatted.txt
-          echo \`\`\` | tee -a checksums-formatted.txt
-          cat checksums.txt | tee -a checksums-formatted.txt
-          echo \`\`\` | tee -a checksums-formatted.txt
+      - name: Run GoReleaser
+        # GoReleaser v4.2.0
```
Comment (praveenrewar) [inline .github/workflows/release.yml:27]: nit: This should also be removed I think?

**Review by praveenrewar — APPROVED** [review]
LGTM!
(Something for for future, we should move the testing steps to a single step as it's all relevant)

File: .github/workflows/release.yml (hunk 1 above)
Comment (rohitagg2020) [reply .github/workflows/release.yml:27]: Since we are using the commit below, it is good to know this commit belongs to which tag.

**Comment by kumaritanushree** [comment]
LGTM

**Pull request merged**